manifests/secrets.pp

```diff
@@ -1,14 +1,34 @@
-# StrongSwan secrets configuration.
-# **$selectors** - list of selectors for current secret(@see ipsec.secrets(5))
+# @summary strongSwan secrets configuration.
+#
+# @param selectors
+#   List of selectors for current secret
+# @param options
+#   A hash of options to use with the secret
+#
+# @example Example 1
+#    strongswan::secrets { '%any':
+#      options => {
+#        'RSA' => 'vpnHostKey.der keypass'
+#      }
+#    }
+#
+# @example Example 2
+#    strongswan::secrets { 'John':
+#      options => {
+#        'EAP' => 'SuperSecretPass'
+#      }
+#    }
+#
+# @see https://linux.die.net/man/5/ipsec.secrets
 define strongswan::secrets(
   Array[String[1]] $selectors = [$title],
   Hash $options               = {},
 ) {
+  require strongswan::package
+
   concat::fragment { "ipsec_secrets_secret-${title}":
     content => template('strongswan/ipsec_secrets_secret.erb'),
     target  => $strongswan::ipsec_secrets,
     order   => '02',
-    require => Class['strongswan'],
-    notify  => Class['strongswan::service'],
```

Comment:
relationship moved to init.pp

Reply:
TODO: Investigate whether the relationship is even required.

Reply:
No, it isn't.  `concat::fragments` are effectively always insync and don't generate events.  It's the `concat` resource that needs to have the relationship (which it already does as it's in `config.pp`).

Reply:
Fixed.